website/src/content/deployment-examples/azure.mdx

```diff
@@ -0,0 +1,103 @@
+---
+title: Deploying to Microsoft Azure with Static Web Apps
+---
+
+This deployment example refers to [Microsoft Azure](https://azure.microsoft.com/).
+
+# Prerequisites
+
+Before you get started, you need to have:
+
+- A [Microsoft Azure](https://azure.microsoft.com/free/) account.
+- A [commercetools](https://docs.commercetools.com/getting-started) account and a Project.
+- A Custom Application [configured in the Merchant Center](https://docs.commercetools.com/merchant-center/managing-custom-applications).
+
+# Configuration
+
+In your Custom Application config, make sure to provide the [Custom Application ID](/api-reference/application-config#envproductionapplicationid) that you got when you configured the Custom Application in the Merchant Center.
+
+Moreover, you need to provide the [production URL](/api-reference/application-config#envproductionurl) from your Microsoft Azure project. You can keep the standard Azure URL `https://<project>.1.azurestaticapps.net` or provide your own custom domain.
+
+```json title="custom-application-config.json"
+{
+  "env": {
+    "production": {
+      "applicationId": "ckvtahxl90097sys6har1e6n3",
+      "url": "https://<project>.1.azurestaticapps.net"
+    }
+  }
+}
+```
+
+## Using environment variables
+
+In case you want to avoid hardcoding certain values, for example the Application ID, or the Project key, you can [use variable placeholders](/api-reference/application-config#using-variable-placeholders) in your Custom Application config.
+
+```json title="custom-application-config.json" highlightLines="4"
+{
+  "env": {
+    "production": {
+      "applicationId": "${env:APPLICATION_ID}",
+      "url": "https://<project>.1.azurestaticapps.net"
+    }
+  }
+}
+```
+
+# Connect Microsoft Azure with GitHub Actions
+
+The easiest way to deploy to Microsoft Azure is to use [Static Web Apps](https://azure.microsoft.com/en-us/services/app-service/static/#overview) deployment service. This service enables first-class GitHub integration.
+
+Follow the steps in the [Microsoft Azure Static Web App creator](https://portal.azure.com/#create/Microsoft.StaticApp) to create a new project and select `GitHub` as the deployment source.
+
+<Info>
+
+Make sure to grant Microsoft Azure access to your repository in the following step.
+
+</Info>
+
+## Configuring build settings
+
+In the Static Web App setup process you need to configure the following things:
+
+* Select your organization, repository and branch.
+* In the **Build details** section select `Custom` as the build preset.
+* Provide the following build settings:
+    * App location: `/`
+    * Output location: `public`
+
+<Info>
+
+If your Custom Application config [requires environment variables](#using-environment-variables), make sure to provide them in your **Home > Static Web Apps > Your project name > Configuration > Application Settings**.
```

Comment:
Is this step needed when using github actions?

Reply:
It is, because the auto-provisioned GH Action is configured to read secrets from the Static Web App env vars config:
https://docs.microsoft.com/en-us/azure/static-web-apps/build-configuration?tabs=github-actions#environment-variables

[This is my example of GH Action](https://github.com/kark/azure-deployment-example-via-portal/blob/95d54f13ac31da0987f9cd7a4981d4736ab67b62/.github/workflows/azure-static-web-apps-red-rock-0ff5a4003.yml#L35) which was created by Azure and those secrets are read from Azure portal config

Reply:
I don't have access to that repo.

Reply:
Sorry, just made it public

Reply:
So when you connect the repo to Azure, Azure creates the github secrets automatically based on the environment variables that you defined in Azure? And then you just reference them in the github action?

From the example in the docs, it seems to me that you can just provide the variables in the github action, without having to define them in Azure as well.

That's where I'm a bit confused 🤔

Reply:
My bad, apparently I remembered it all wrong, sorry for all the confusion 😕🤦‍♂️ I remember trying out various options of env vars settings some time ago when I configured that.

Anyhow, you are totally right, the secrets are read from github repo settings and the whole section `If your Custom Application config [requires environment variables](#using-environment-variables), make sure to provide them in your **Home > Static Web Apps > Your project name > Configuration > Application Settings**.` is not needed.